components/style/properties/properties.mako.rs

```diff
@@ -287,14 +331,57 @@ impl Clone for PropertyDeclaration {
     fn clone(&self) -> Self {
         use self::PropertyDeclaration::*;
 
+        <%
+            [copy, others] = [list(g) for _, g in groupby(variants, key=lambda x: not x["copy"])]
+        %>
+
+        let self_tag = unsafe {
+            (*(self as *const _ as *const PropertyDeclarationVariantRepr<()>)).tag
+        };
+        if self_tag <= LonghandId::${copy[-1]["name"]} as u16 {
+            #[derive(Clone, Copy)]
+            #[repr(u16)]
+            enum CopyVariants {
+                % for v in copy:
+                _${v["name"]}(${v["type"]}),
+                % endfor
+            }
+
+            unsafe {
+                let mut out = mem::uninitialized();
+                ptr::write(
+                    &mut out as *mut _ as *mut CopyVariants,
+                    *(self as *const _ as *const CopyVariants),
+                );
+                return out;
+            }
+        }
+
         match *self {
-            % for ty, variants in groups.iteritems():
-            % if len(variants) == 1:
-            ${variants[0]}(ref value) => {
-                ${variants[0]}(value.clone())
+            ${" |\n".join("{}(..)".format(v["name"]) for v in copy)} => {
+                #[cfg(debug_assertions)]
+                unreachable!();
```

Comment:
Can you comment on why the `debug_unreachable` crate didn't cut it?

Reply:
`debug_unreachable` actually does the exact same thing so I'm switching that PR to it.